src/figuresClasses.ts

```diff
@@ -2,10 +1,88 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  sides: { a: number; b: number; c: number };
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        'The longest side of a triangle must be less than a sum of two others',
+      );
+    }
+    this.color = color;
+    this.sides = { a, b, c };
+  }
+
+  getArea(): number {
+    const { a, b, c } = this.sides;
+    const p = (a + b + c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
+
+  color: 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+  radius: number;
 
-export class Circle implements Figure {}
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    if (radius <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
 
-export class Rectangle implements Figure {}
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  width: number;
+
+  height: number;
+
+  constructor(color: 'red' | 'green' | 'blue', width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    this.color = color;
+    this.height = height;
+    this.width = width;
+  }
+
+  getArea(): number {
+    return Math.floor(this.height * this.width * 100) / 100;
+  }
+}
 
-export function getInfo(figure): string {
-  return typeof figure;
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
```

Comment:
Ensure that the area is formatted to two decimal places in the `getInfo` function, similar to how it's done in the `getArea` methods of the classes.